File: 01a_File_Formats_Bonanza/FileReaderTypescript/src/LoadPersonFrom.ts
import { readFileSync } from 'fs';
import { Person } from './Person';
import YAML from 'yaml'

export class LoadPersonFrom {
    // static Xml(fileLocation: string): Person {
    //     let jsonData;
    //     var data = readFileSync(fileLocation, 'utf8');
    //     parseString(data, function(err, result){
    //         jsonData = result;
    //     })
    //     console.log(jsonData);
    //     jsonData = JSON.stringify(jsonData)        
    //     console.log(jsonData);
    //     return jsonData as Person
    // }

    // static Csv(fileLocation: string): Person {
    //     var data = readFileSync(fileLocation, 'utf8');        
    //     return CSV.parse(data) as Person
    // }

    static Yaml(fileLocation: string): Person {
        var data = readFileSync(fileLocation, 'utf8');
        return YAML.parse(data) as Person;
    }

    static Json(fileLocation: string): Person {
        var data = readFileSync(fileLocation, 'utf8');
        return JSON.parse(data) as Person
    }
}

File: 01a_File_Formats_Bonanza/FileReaderDotnet/LoadPersonFrom.cs
using Newtonsoft.Json;
using CsvHelper;
using System.Globalization;
using System.Xml.Serialization;
using YamlDotNet.Serialization;
using YamlDotNet.Serialization.NamingConventions;

public static class LoadPersonFrom
{
    public static Person Yaml(string fileLocation)
    {
        var deserializer = new DeserializerBuilder().WithNamingConvention(CamelCaseNamingConvention.Instance).Build();
        using (StreamReader reader = File.OpenText(fileLocation))
        {            
            Person person = deserializer.Deserialize<Person>(reader);
            return person;
        }
    }

    public static Person Xml(string fileLocation)
    {
        var serializer = new XmlSerializer(typeof(Person));
        using (var reader = new StreamReader(fileLocation))
        {
            Person person = (Person)serializer.Deserialize(reader);
            return person;
        }
    }

    public static Person Csv(string fileLocation)
    {
        using StreamReader reader = new StreamReader(fileLocation);
        using CsvReader csv = new CsvReader(reader, CultureInfo.InvariantCulture);
        csv.Context.RegisterClassMap<PersonMap>();
        var records = csv.GetRecords<Person>();
        Person person = records.First();
        //manually convert string array of hobbies to true list of hobbies
        List<string> hobbies = string.Join(",", person.hobbies).Split(",").ToList();
        person.hobbies = hobbies;
        return person;
    }

    public static Person Json(string fileLocation)
    {
        var serializer = new JsonSerializer();
        using (StreamReader reader = File.OpenText(fileLocation))
        {
            Person person = (Person)serializer.Deserialize(reader, typeof(Person));
            return person;
        }
    }
}